src/zarr/api/asynchronous.py

```diff
@@ -724,7 +724,7 @@ async def create(
     if meta_array is not None:
         warnings.warn("meta_array is not yet implemented", RuntimeWarning, stacklevel=2)
 
-    mode = kwargs.pop("mode", cast(AccessModeLiteral, "r" if read_only else "w"))
+    mode = kwargs.pop("mode", cast(AccessModeLiteral, "r" if read_only else "a"))
     store_path = await make_store_path(store, mode=mode)
```

Comment:
This logic could use a second thought. If the store is already instantiated with a `mode`, there is no need for a default. We should just make sure `mode is None` and that `read_only` is compatible with the state of the store.